### PR TITLE
Make entire address and ENS name labels tappable to trigger the copy action

### LINKED ISSUE #1806: Make entire address and ENS name labels tappable to trigger the copy action
Currently only the blue icon is tappable. Note that the screenshot doesn't show the ens name label. Look around `RequestViewController.copyEnsButton`. The screen is accessible from Settings > My Wallet Address.

Simulator Screen Shot - iPhone X - 2020-05-19 at 23 20 14

### PR DESCRIPTION
**Pull request by vladyslav-iosdev**
Closes #1806

Make entire address and ENS name labels tappable to trigger the copy action
Add tap gesture recogizer for ens and address labels

### CODE REVIEW

**Comment by hboon** [comment]
Would you make the commit message be:

“Make entire address and ENS name labels tappable to trigger the copy action #1806”?

**Comment by hboon** [comment]
But yes, this is the single commit I was referring to. Thanks

**Comment by vladyslav-iosdev** [comment]
> Make entire address and ENS name labels tappable to trigger the copy action #1806”?

I'd do it, but i don't know how

**Comment by hboon** [comment]
I’ll paste the steps later. Don’t worry

**Comment by hboon** [comment]
@vladyslav-iosdev 

```
git checkout "#1806"
git commit --amend -m "Make entire address and ENS name labels tappable to trigger the copy action #1806"
git push origin "#1806" --force #likely to be "origin". The remote which points to your version of the repo.
```

#git commit --amend changes the last commit

**Comment by hboon** [comment]
The reason to keep it a 1-liner is so it is very easy to print out the git logs (and read or grep through them). After that first line, usually, we'll list the reason for the change if it's not obvious, rather than the details of the changes themselves because the changes can usually be inferred from the changed files themselves. Unless it's a very big change.

**Comment by vladyslav-iosdev** [comment]
> @vladyslav-iosdev
> 
> ```
> git checkout "#1806"
> git commit --amend -m "Make entire address and ENS name labels tappable to trigger the copy action #1806"
> git push origin "#1806" --force #likely to be "origin". The remote which points to your version of the repo.
> ```
> 
> #git commit --amend changes the last commit

this one should be done

**Comment by hboon** [comment]
I have these remotes:

$ git remote -v

origin  git@github.com:AlphaWallet/alpha-wallet-ios.git (fetch) #probably your `upstream`
origin  git@github.com:AlphaWallet/alpha-wallet-ios.git (push)
vlad    git@github.com:vladyslav-iosdev/alpha-wallet-ios.git (fetch) #probably your `origin`
vlad    git@github.com:vladyslav-iosdev/alpha-wallet-ios.git (push)

You will probably have different names, probably upstream and origin respectively. Note that your `origin` and my `origin` likely points to different repo.

Here's what I did (using my remotes):

```
git checkout #1806
git fetch origin master:master #gets the updated `master` and updates it locally
git rebase master #Apply the commits on this branch over `master` and update current branch
git log:

3a7e120c110a50b2720fca6c60f8df70cb70820b Make entire address and ENS name labels tappable to trigger the copy action #1806
1e5a20734882675cf6a59fd96e558f2f42df36a7 Merge pull request #1800 from AlphaWallet/upgrade-schema-2020-06
143670bed6da66ca1c2c62e916409337e6b12c37 update and strengthen tests with new schema 2020-06
3d52fb8e57062cf75a9648ad55b7c923aa98c297 Merge pull request #1813 from AlphaWallet/origin/readme-update
```

Notice that there's a new commit `3a7e120c110a50b2720fca6c60f8df70cb70820b` generated and the next commit `1e5a20734882675cf6a59fd96e558f2f42df36a7` is the tip of `master`. This is usually what we want to have, new commits on top of the tip of `master`, in the feature branch for a PR.

git push vlad "#1806" --force #force needed because we have re-written history with the rebase command
```


I have forced pushed this version into your forked repo.

Before this, this feature branch was:

```
commit 3f70981afd828323afae8c6f0158208125f85d22 (HEAD -> #1806)
Merge: 3f04b956b 000ad23b4
Author: Vladyslav shepitko <vlad_shepitko@outlook.com>
Date:   2020-05-20 19:42:05 +0300

    Merge branch '#1806' of https://github.com/vladyslav-iosdev/alpha-wallet-ios into #1806

commit 3f04b956b9362e8669b786f50644ae255204316a
Author: Vladyslav shepitko <vlad_shepitko@outlook.com>
Date:   2020-05-20 15:19:46 +0300

    Make entire address and ENS name labels tappable to trigger the copy action #1806
```

We generally, wouldn't want that Merge commit there. (the `git rebase master` commit took care of it)

**Comment by hboon** [comment]
Since I have rewritten the #1806 branch in your repo, if you need to get it (example to make changes to it for the PR), you have to fetch them:

```
git fetch vlad "#1806" #remember this is my remote setup. your remote name is probably "origin"
git checkout #1806
git reset --hard vlad/#1806 #this forces your local #1806 to be the same as the one you fetch 2 commands ago
#work and then push. --force if you rewrite history
```

**Comment by hboon** [comment]
@vladyslav-iosdev will review. Thanks

File: AlphaWallet/Transfer/ViewControllers/RequestViewController.swift [hunk 1]
```diff
@@ -74,7 +78,10 @@ class RequestViewController: UIViewController {
 
 		view.backgroundColor = viewModel.backgroundColor
 		view.addSubview(roundedBackground)
-
+        
+        ensLabel.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(copyEns)))
```
Comment (hboon) [inline AlphaWallet/Transfer/ViewControllers/RequestViewController.swift:82]: Sorry, didn't realise the containers (the rounded rects) are different from the labels. Can you add the gesture recognisers to addressContainerView and ensContainerView instead so the hit test area is bigger?

**Comment by vladyslav-iosdev** [comment]
> Since I have rewritten the branch `#1806` in your repo, if you need to get it (example to make changes to it for the PR), you have to fetch them:
> 
> ```
> git fetch vlad "#1806" #remember this is my remote setup. your remote name is probably "origin"
> git checkout #1806
> git reset --hard vlad/#1806 #this forces your local #1806 to be the same as the one you fetch 2 commands ago
> #work and then push. --force if you rewrite history
> ```

Hi, i did it, but for now it seems like i cant run a project.

**Comment by hboon** [comment]
Can you verify if the commit on your development machine matches this? : https://github.com/AlphaWallet/alpha-wallet-ios/pull/1815/commits

**Comment by vladyslav-iosdev** [comment]
> https://github.com/AlphaWallet/alpha-wallet-ios/pull/1815/commits

seems like no

**Comment by hboon** [comment]
If you have 3a7e120c110a50b2720fca6c60f8df70cb70820b, it is the right commit. Try close Xcode, pod install and open it again?

**Comment by vladyslav-iosdev** [comment]
> If you have [3a7e120](https://github.com/AlphaWallet/alpha-wallet-ios/commit/3a7e120c110a50b2720fca6c60f8df70cb70820b), it is the right commit. Try close Xcode, pod install and open it again?

Thanks! now its ok.

File: AlphaWallet/Transfer/ViewControllers/RequestViewController.swift (hunk 1 above)
Comment (vladyslav-iosdev) [reply AlphaWallet/Transfer/ViewControllers/RequestViewController.swift]: @hboon file was removed, sent changes.

**Comment by hboon** [comment]
@vladyslav-iosdev notice the closes (or fixes if it's a bug) keyword used in the 1st comment of this PR.

**Pull request merged**